tests/test_response_media.py

```diff
@@ -0,0 +1,106 @@
+import pytest
+import six
+
+import falcon
+from falcon import errors, media, testing
+
+
+def create_client(handlers=None):
+    res = testing.SimpleTestResource()
+
+    app = falcon.API()
+    app.add_route('/', res)
+
+    if handlers:
+        app.resp_options.media_handlers.update(handlers)
+
+    client = testing.TestClient(app)
+    client.resource = res
+
+    return client
+
+
+@pytest.mark.parametrize('media_type', [
+    ('*/*'),
+    ('application/json'),
+    ('application/json; charset=utf-8'),
+])
+def test_json(media_type):
+    client = create_client()
+    client.simulate_get('/')
+
+    resp = client.resource.captured_resp
+    resp.content_type = media_type
+    resp.media = {'something': True}
+
+    assert resp.data == b'{"something": true}'
+
+
+@pytest.mark.parametrize('media_type', [
+    ('application/msgpack'),
+    ('application/msgpack; charset=utf-8'),
+    ('application/x-msgpack'),
+])
+def test_msgpack(media_type):
+    client = create_client({
+        'application/msgpack': media.MessagePackHandler(),
+        'application/x-msgpack': media.MessagePackHandler(),
+    })
+    client.simulate_get('/')
+
+    resp = client.resource.captured_resp
+    resp.content_type = media_type
+    resp.media = {b'something': True}
```

Comment:
[fix] Test an object that contains both `unicode` and `bytes` types to ensure the MessagePack bin type is in play.